core/src/main/java/com/linecorp/armeria/internal/common/AbstractKeepAliveHandler.java

```diff
@@ -120,14 +127,22 @@ protected AbstractKeepAliveHandler(Channel channel, String name, Timer keepAlive
     public final void initialize(ChannelHandlerContext ctx) {
```

Comment:
I believe this method is called when the `HttpSessionHandler` is added and not necessarily when the connection is ready for sending/receiving requests.